internal/lightcone/abundance/perfecttiming/perfecttiming.go

```diff
@@ -0,0 +1,60 @@
+package perfecttiming
+
+import (
+	"github.com/simimpact/srsim/pkg/engine"
+	"github.com/simimpact/srsim/pkg/engine/equip/lightcone"
+	"github.com/simimpact/srsim/pkg/engine/info"
+	"github.com/simimpact/srsim/pkg/engine/modifier"
+	"github.com/simimpact/srsim/pkg/engine/prop"
+	"github.com/simimpact/srsim/pkg/key"
+	"github.com/simimpact/srsim/pkg/model"
+)
+
+// Increases the wearer's Effect RES by 16% and increases Outgoing Healing by an amount
+// that is equal to 33% of Effect RES. Outgoing Healing can be increased this way by up to 15%.
+
+const (
+	PTEffRes    key.Modifier = "perfect_timing_eff_res"
+	PTHealBoost key.Modifier = "perfect_timing_heal_boost"
+)
+
+func init() {
+	lightcone.Register(key.PerfectTiming, lightcone.Config{
+		CreatePassive: Create,
+		Rarity:        4,
+		Path:          model.Path_ABUNDANCE,
+		Promotions:    promotions,
+	})
+	modifier.Register(PTEffRes, modifier.Config{
+		Listeners: modifier.Listeners{
+			OnAdd: giveHealBuff,
+		},
+	})
+	modifier.Register(PTHealBoost, modifier.Config{})
+}
+
+func Create(engine engine.Engine, owner key.TargetID, lc info.LightCone) {
+	effResAmt := 0.12 + 0.04*float64(lc.Imposition)
+	engine.AddModifier(owner, info.Modifier{
+		Name:   PTEffRes,
+		Source: owner,
+		Stats:  info.PropMap{prop.EffectRES: effResAmt},
+		State:  float64(lc.Imposition), // 2 calcs outside depends on lc imp. level.
+	})
+}
```

Comment:
Here is my suggestion on how to make this match the game behavior:
1. Remove all listeners from `PTEffRes` and the `State` (so this modifier now only adds `EffectRES` and does not manage healing boost)
2. Create a `giveHealBuff` method like you currently have, but make the signature `giveHealBuff(engine engine.Engine, owner *info.Stats, imposition int, prevBuff float64) float64`
  - this new method should calculate the heal buff amount like it currently does
  - if the calculated amount `!= prevBuff` that is passed in
    - call `AddModifier` and populate the `Stats` line with the new buff amount
    - return the new buff amount
  - if the calculated amount `== prevBuff`, do nothing and return the prevBuff amount (only add the modifier if the buff amount changes)
4. Create a `BattleStart` subscription that will call `giveHealBuff` on the owner target. Example of how to do this: https://github.com/simimpact/srsim/blob/6643c1b61df8bf8ba25064a8bbaa1a6b4f305b34/internal/relic/planar/vonwacq/vonwacq.go#L32-L38
5. Create a `ActionEnd` subscription that will call `engine.Stats` for the owner and then call `giveHealBuff`

A pseudo version of what this `Create` method might look like:
```go
func Create(eng engine.Engine, owner key.TargetID, lc info.LightCone) {
	effResAmt := 0.12 + 0.04*float64(lc.Imposition)
  healBuffAmt := 0

  // add the effect res buff
	engine.AddModifier(owner, info.Modifier{
		Name:   PTEffRes,
		Source: owner,
		Stats:  info.PropMap{prop.EffectRES: effResAmt},
	})

  engine.Events().BattleStart.Subscribe(func(e event.BattleStart) {
		for _, char := range e.CharStats {
			if char.ID() == owner {
				healBuffAmt = giveHealBuff(eng, char, lc.Imposition, healBuffAmt)
			}
		}
	})

  engine.Events().ActionEnd.Subscribe(func(e event.ActionEnd) {
    healBuffAmt = giveHealBuff(eng, eng.Stats(owner), lc.Imposition, healBuffAmt)
  })
}
```